### PR TITLE
tidb,executor: avoid get startTS for read operation in READ COMMITTED isolation level

### PR DESCRIPTION
**Pull request by tiancaiamao**
In READ COMMITTED isolation level, unrepeatable read anomaly is allowed,
so we can always use math.MaxUint64 to read the latest committed value,
it would decrease the latency for read-only transaction.

PTAL @disksing  @coocood  @wentaoxu  @shenli 

It should beneficial to AP query @ilovesoup

### CODE REVIEW

**Comment by shenli** [comment]
@tiancaiamao Use MaxUint64 will meet trouble when a DDL is running.

**Comment by tiancaiamao** [comment]
Oh, I remember it, thanks for remind... @shenli 
I'll take a more closely look tomorrow.

File: executor/adapter.go [hunk 1]
```diff
@@ -335,3 +336,36 @@ func IsPointGetWithPKOrUniqueKeyByAutoCommit(ctx context.Context, p plan.Plan) b
...
 }
+
+// isReadCommittedGet returns true when it's not write operation and in READ COMMITTED isolation level.
+func isReadCommittedGet(ctx context.Context, p plan.Plan) bool {
+	if ctx.GetSessionVars().Systems[variable.TxnIsolation] != ast.ReadCommitted {
+		return false
+	}
+
+	// check plan
+	if proj, ok := p.(*plan.Projection); ok {
```
Comment (coocood) [inline executor/adapter.go]: How about check `*plan.Update, *plan.Delete, *plan.Insert` and return false, otherwise, return true?

Comment (coocood) [reply executor/adapter.go]: And there is `*plan.Execute` that executes another plan.

Comment (wentaoxu) [reply executor/adapter.go]: should ```*plan.anaylze``` be added here?

Comment (tiancaiamao) [reply executor/adapter.go]: I think we can be more conservative at first. @coocood

**Comment by tiancaiamao** [comment]
testStatsCacheSuite in unit test fail, but I don't know how to fix it.

**Pull request closed without merging**